File: rxbinding/src/main/java/com/jakewharton/rxbinding2/widget/RxProgressBar.java
package com.jakewharton.rxbinding2.widget;

import androidx.annotation.CheckResult;
import androidx.annotation.NonNull;
import android.widget.ProgressBar;
import io.reactivex.functions.Consumer;

import static com.jakewharton.rxbinding2.internal.Preconditions.checkNotNull;

public final class RxProgressBar {
  /**
   * An action which increments the progress value of {@code view}.
   * <p>
   * <em>Warning:</em> The created observable keeps a strong reference to {@code view}. Unsubscribe
   * to free this reference.
   *
   * @deprecated Use view::incrementProgressBy method reference.
   */
  @Deprecated
  @CheckResult @NonNull
  public static Consumer<? super Integer> incrementProgressBy(@NonNull ProgressBar view) {
    checkNotNull(view, "view == null");
    return view::incrementProgressBy;
  }

  /**
   * An action which increments the secondary progress value of {@code view}.
   * <p>
   * <em>Warning:</em> The created observable keeps a strong reference to {@code view}. Unsubscribe
   * to free this reference.
   *
   * @deprecated Use view::incrementSecondaryProgressBy method reference.
   */
  @Deprecated
  @CheckResult @NonNull
  public static Consumer<? super Integer> incrementSecondaryProgressBy(@NonNull ProgressBar view) {
    checkNotNull(view, "view == null");
    return view::incrementSecondaryProgressBy;
  }

  /**
   * An action which sets whether {@code view} is indeterminate.
   * <p>
   * <em>Warning:</em> The created observable keeps a strong reference to {@code view}. Unsubscribe
   * to free this reference.
   *
   * @deprecated Use view::setIndeterminate method reference.
   */
  @Deprecated
  @CheckResult @NonNull
  public static Consumer<? super Boolean> indeterminate(@NonNull ProgressBar view) {
    checkNotNull(view, "view == null");
    return view::setIndeterminate;
  }

  /**
   * An action which sets the max value of {@code view}.
   * <p>
   * <em>Warning:</em> The created observable keeps a strong reference to {@code view}. Unsubscribe
   * to free this reference.
   *
   * @deprecated Use view::setMax method reference.
   */
  @Deprecated
  @CheckResult @NonNull
  public static Consumer<? super Integer> max(@NonNull ProgressBar view) {
    checkNotNull(view, "view == null");
    return view::setMax;
  }

  /**
   * An action which sets the progress value of {@code view}.
   * <p>
   * <em>Warning:</em> The created observable keeps a strong reference to {@code view}. Unsubscribe
   * to free this reference.
   *
   * @deprecated Use view::setProgress method reference.
   */
  @Deprecated
  @CheckResult @NonNull
  public static Consumer<? super Integer> progress(@NonNull ProgressBar view) {
    checkNotNull(view, "view == null");
    return view::setProgress;
  }

  /**
   * An action which sets the secondary progress value of {@code view}.
   * <p>
   * <em>Warning:</em> The created observable keeps a strong reference to {@code view}. Unsubscribe
   * to free this reference.
   *
   * @deprecated Use view::setSecondaryProgress method reference.
   */
  @Deprecated
  @CheckResult @NonNull
  public static Consumer<? super Integer> secondaryProgress(@NonNull ProgressBar view) {
    checkNotNull(view, "view == null");
    return view::setSecondaryProgress;
  }

  private RxProgressBar() {
    throw new AssertionError("No instances.");
  }
}

File: rxbinding-kotlin/src/main/kotlin/com/jakewharton/rxbinding2/widget/RxProgressBar.kt
@file:Suppress("NOTHING_TO_INLINE")

package com.jakewharton.rxbinding2.widget

import android.support.annotation.CheckResult
import android.widget.ProgressBar
import io.reactivex.functions.Consumer
import kotlin.Deprecated
import kotlin.Int
import kotlin.Suppress

/**
 * An action which increments the progress value of `view`.
 *
 * *Warning:* The created observable keeps a strong reference to `view`. Unsubscribe
 * to free this reference.
 */
@Deprecated("Use view::incrementProgressBy method reference.")
@CheckResult
inline fun ProgressBar.incrementProgressBy(): Consumer<in Int> = RxProgressBar.incrementProgressBy(this)

/**
 * An action which increments the secondary progress value of `view`.
 *
 * *Warning:* The created observable keeps a strong reference to `view`. Unsubscribe
 * to free this reference.
 */
@Deprecated("Use view::incrementSecondaryProgressBy method reference.")
@CheckResult
inline fun ProgressBar.incrementSecondaryProgressBy(): Consumer<in Int> = RxProgressBar.incrementSecondaryProgressBy(this)

/**
 * An action which sets whether `view` is indeterminate.
 *
 * *Warning:* The created observable keeps a strong reference to `view`. Unsubscribe
 * to free this reference.
 */
@Deprecated("Use view::setIndeterminate method reference.")
@CheckResult
inline fun ProgressBar.indeterminate(): Consumer<in Boolean> = RxProgressBar.indeterminate(this)

/**
 * An action which sets the max value of `view`.
 *
 * *Warning:* The created observable keeps a strong reference to `view`. Unsubscribe
 * to free this reference.
 */
@Deprecated("Use view::setMax method reference.")
@CheckResult
inline fun ProgressBar.max(): Consumer<in Int> = RxProgressBar.max(this)

/**
 * An action which sets the progress value of `view`.
 *
 * *Warning:* The created observable keeps a strong reference to `view`. Unsubscribe
 * to free this reference.
 */
@Deprecated("Use view::setProgress method reference.")
@CheckResult
inline fun ProgressBar.progress(): Consumer<in Int> = RxProgressBar.progress(this)

/**
 * An action which sets the secondary progress value of `view`.
 *
 * *Warning:* The created observable keeps a strong reference to `view`. Unsubscribe
 * to free this reference.
 */
@Deprecated("Use view::setSecondaryProgress method reference.")
@CheckResult
inline fun ProgressBar.secondaryProgress(): Consumer<in Int> = RxProgressBar.secondaryProgress(this)
